src/mmw/js/src/data_catalog/models.js

```diff
@@ -29,10 +28,30 @@ var Catalog = Backbone.Model.extend({
         error: '',
     },
 
-    search: function(query, fromDate, toDate, bounds) {
+    searchIfNeeded: function(query, fromDate, toDate, bbox) {
+        var self = this,
+            isSameSearch = query === this.get('query') &&
+                           fromDate === this.get('fromDate') &&
+                           toDate === this.get('toDate') &&
+                           bbox === this.get('bbox');
+
+        if (!isSameSearch) {
+            if (this.searchPromise) {
+                this.searchPromise.reject({ superceded: true });
```

Comment:
Actually, this isn't going to work. `this.searchPromise` isn't a deferred object. It's a `jqXHR` because that's what backbone returns with `.fetch`. I think I'm going to change this to `abort`. Another option is to wrap the `fetch` in an deferred , but I think that makes it a little more difficult to follow.